### PR TITLE
`innodb_autoinc_lock_mode` tests

### PR DESCRIPTION
**Pull request by jycor**
companion pr: https://github.com/dolthub/go-mysql-server/pull/2410

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@jycor __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>765f27c</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>765f27c</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

File: integration-tests/bats/sql-server.bats [hunk 1]
```diff
@@ -25,6 +25,39 @@ teardown() {
     teardown_common
 }
 
+@test "sql-server: innodb_autoinc_lock_mode is set to 2" {
```
Comment (max-hoffman) [inline integration-tests/bats/sql-server.bats:28]: noted in GMS PR that this pattern might be useful for enginetests

**Review by max-hoffman — APPROVED** [review]
lgtm

**Comment by coffeegoddd** [comment]
@jycor __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>69bcbf1</TD>
<TD>ok</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>69bcbf1</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

**Pull request merged**